tests/5.0/target/test_target_collapse.c

```diff
@@ -0,0 +1,62 @@
+//===------ test_target_collapse.c --------------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+// Tests the target directive with collapse clause.
+//
+//===------------------------------------------------------------------------===//
+
+
+#include <stdio.h>
+#include "omp.h"
+#include "ompvv.h"
+
+#define N 5
+int Runtst(int gpu) {
+  int A[N], B[N] = {0};
+  int errors = 0;
+  for (int i = 0; i < N; ++i) {
+    A[i] = 0;
+    B[i] = i + 1;
+  }
+  int ThrdTrack[N * N + N] = {0}; // an array of 30 elements
+#pragma omp target data map(tofrom: A, B, ThrdTrack) device(gpu)
+#pragma omp target parallel for collapse(2) shared(A, B, ThrdTrack) device(gpu)
+  for (int i = 0; i < N; ++i) {
+    for (int j = 0; j < N; ++j) {
+      int ThrdId = omp_get_thread_num();
+      ThrdTrack[ThrdId] = ThrdId;
```

Comment:
```suggestion

```

Reply:
It is best to move away from thread num specific checking,